lib/user_space/user_aarch64_atomic.h

```diff
@@ -0,0 +1,334 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2012-2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#pragma once
+
+#ifndef __MOTR_LIB_USER_AARCH64_ATOMIC_H__
+#define __MOTR_LIB_USER_AARCH64_ATOMIC_H__
+
+#include "lib/types.h"
+#include "lib/assert.h"
+
+/**
+   @addtogroup atomic
+
+   Implementation of atomic operations for Linux user space uses x86_64/aarch64 
+   assembly language instructions (with gcc syntax). "Lock" prefix is used
+   everywhere for x86_64 platform whereas the aarch64 uses its own set of atomic 
+   assembely instruction to ensure atomicity ---no optimisation for non-SMP 
+   configurations in present.
+ */
+
+struct m0_atomic64 {
+	long a_value;
+};
+
+static inline void m0_atomic64_add(struct m0_atomic64 *a, int64_t num);
+static inline void m0_atomic64_sub(struct m0_atomic64 *a, int64_t num);
+
+static inline void m0_atomic64_set(struct m0_atomic64 *a, int64_t num)
+{
+	M0_CASSERT(sizeof a->a_value == sizeof num);
+
+	a->a_value = num;
+}
+
+/**
+   Returns value of an atomic counter.
+ */
+static inline int64_t m0_atomic64_get(const struct m0_atomic64 *a)
+{
+	return a->a_value;
+}
+
+/**
+ atomically increment counter
+
+ @param a pointer to atomic counter
+
+ @return none
+ */
+static inline void m0_atomic64_inc(struct m0_atomic64 *a)
+{
+#ifdef CONFIG_X86_64
+	asm volatile("lock incq %0"
+		     : "=m" (a->a_value)
+		     : "m" (a->a_value));
+#elif defined CONFIG_AARCH64
+	m0_atomic64_add(a, (int64_t)1);
+#else
+#error  "The Platform is not supported"
+#endif
```

Comment:
Since this file is platform-specific - there is no need to check for `CONFIG_X86_64` or any other platform. Please remove here and in other places.

Reply:
Removed